etcdserver/server.go

```diff
@@ -147,9 +147,14 @@ func NewServer(cfg *ServerConfig) (*EtcdServer, error) {
 	if err != nil {
 		return nil, err
 	}
-	if walVersion == wal.WALUnknown {
+	switch walVersion {
+	case wal.WALUnknown:
 		return nil, fmt.Errorf("unknown wal version in data dir %s", cfg.DataDir)
+	case wal.WALv2_0Proxy:
+		// skip the discovery path and return the previous full cluster error again.
+		return nil, discovery.ErrFullCluster
```

Comment:
I think it is better to use a separate error. It is unclear about why wal.WALv2_0Proxy leads to discovery.ErrFullCluster. Considering user who bootstrap etcd member from the same data dir which is used for proxy before, the error `discovery.ErrFullCluster` is weird.